streams/src/test/java/org/apache/kafka/streams/kstream/internals/KTableAggregateTest.java

```diff
@@ -73,46 +58,54 @@ public void testAggBasic() {
             .aggregate(
                 MockInitializer.STRING_INIT,
                 MockAggregator.TOSTRING_ADDER,
                 MockAggregator.TOSTRING_REMOVER,
                 Materialized.<String, String, KeyValueStore<Bytes, byte[]>>as("topic1-Canonized")
                     .withValueSerde(stringSerde));
 
+        final MockProcessorSupplier<String, Object> supplier = new MockProcessorSupplier<>();
         table2.toStream().process(supplier);
 
-        driver.setUp(builder, stateDir, Serdes.String(), Serdes.String());
-
-        driver.process(topic1, "A", "1");
-        driver.flushState();
-        driver.process(topic1, "B", "2");
-        driver.flushState();
-        driver.process(topic1, "A", "3");
-        driver.flushState();
-        driver.process(topic1, "B", "4");
-        driver.flushState();
-        driver.process(topic1, "C", "5");
-        driver.flushState();
-        driver.process(topic1, "D", "6");
-        driver.flushState();
-        driver.process(topic1, "B", "7");
-        driver.flushState();
-        driver.process(topic1, "C", "8");
-        driver.flushState();
-
-        assertEquals(
-            asList(
-                "A:0+1 (ts: 0)",
-                "B:0+2 (ts: 0)",
-                "A:0+1-1+3 (ts: 0)",
-                "B:0+2-2+4 (ts: 0)",
-                "C:0+5 (ts: 0)",
-                "D:0+6 (ts: 0)",
-                "B:0+2-2+4-4+7 (ts: 0)",
-                "C:0+5-5+8 (ts: 0)"),
-            supplier.theCapturedProcessor().processed);
+        try (
+            final TopologyTestDriver driver = new TopologyTestDriver(
+                builder.build(),
+                mkProperties(mkMap(
+                    mkEntry(StreamsConfig.BOOTSTRAP_SERVERS_CONFIG, "dummy"),
+                    mkEntry(StreamsConfig.APPLICATION_ID_CONFIG, "test"),
+                    mkEntry(StreamsConfig.STATE_DIR_CONFIG, TestUtils.tempDirectory("kafka-test").getAbsolutePath())
+                )),
+                0L)) {
+            final ConsumerRecordFactory<String, String> recordFactory =
+                new ConsumerRecordFactory<>(new StringSerializer(), new StringSerializer(), 0L, 0L);
+
+            driver.pipeInput(recordFactory.create(topic1, "A", "1"));
+            driver.pipeInput(recordFactory.create(topic1, "B", "2"));
+            driver.pipeInput(recordFactory.create(topic1, "A", "3"));
+            driver.pipeInput(recordFactory.create(topic1, "B", "4"));
+            driver.pipeInput(recordFactory.create(topic1, "C", "5"));
+            driver.pipeInput(recordFactory.create(topic1, "D", "6"));
+            driver.pipeInput(recordFactory.create(topic1, "B", "7"));
+            driver.pipeInput(recordFactory.create(topic1, "C", "8"));
+
+            assertEquals(
+                asList(
+                    "A:0+1 (ts: 0)",
+                    "B:0+2 (ts: 0)",
+                    "A:0+1-1 (ts: 0)",
+                    "A:0+1-1+3 (ts: 0)",
+                    "B:0+2-2 (ts: 0)",
```

Comment:
Same here.

Why do we get more intermediate result?

Reply:
Same explanation, I think.